subsys/net/ip/net_if.c

```diff
@@ -412,6 +414,221 @@ static u8_t get_ipaddr_diff(const u8_t *src, const u8_t *dst, int addr_len)
 
 	return len;
 }
+
+static struct net_if_router *iface_router_lookup(struct net_if *iface,
+						 u8_t family, void *addr)
+{
+	int i;
+
+	for (i = 0; i < CONFIG_NET_MAX_ROUTERS; i++) {
+		if (!routers[i].is_used ||
+		    routers[i].address.family != family ||
+		    routers[i].iface != iface) {
+			continue;
+		}
+
+		if ((IS_ENABLED(CONFIG_NET_IPV6) && family == AF_INET6 &&
+		     net_ipv6_addr_cmp(net_if_router_ipv6(&routers[i]),
+				       (struct in6_addr *)addr)) ||
+		    (IS_ENABLED(CONFIG_NET_IPV6) && family == AF_INET &&
+		     net_ipv4_addr_cmp(net_if_router_ipv4(&routers[i]),
+				       (struct in_addr *)addr))) {
+			return &routers[i];
+		}
+	}
+
+	return NULL;
+}
+
+static void iface_router_run_timer(u32_t current_time)
+{
+	struct net_if_router *router, *next;
+	u32_t new_timer = UINT_MAX;
+
+	if (k_delayed_work_remaining_get(&router_timer)) {
+		k_delayed_work_cancel(&router_timer);
+	}
+
+	SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&active_router_timers,
+					 router, next, node) {
+		new_timer = MIN(router->life_start +
```

Comment:
Avoid arithmetic things in MIN() and MAX() macros, they might have hard to see interactions (I am not saying there is here)